raiden/storage/sqlite.py

```diff
@@ -180,22 +233,62 @@ def __init__(self, database_path: Path):
         with conn:
             conn.executescript(DB_SCRIPT_CREATE_TABLES)
 
-        # When writting to a table where the primary key is the identifier and we want
-        # to return said identifier we use cursor.lastrowid, which uses sqlite's last_insert_rowid
-        # https://github.com/python/cpython/blob/2.7/Modules/_sqlite/cursor.c#L727-L732
-        #
-        # According to the documentation (http://www.sqlite.org/c3ref/last_insert_rowid.html)
-        # if a different thread tries to use the same connection to write into the table
-        # while we query the last_insert_rowid, the result is unpredictable. For that reason
-        # we have this write lock here.
-        #
-        # TODO (If possible):
-        # Improve on this and find a better way to protect against this potential race
-        # condition.
         self.conn = conn
-        self.write_lock = threading.Lock()
```

Comment:
This is one of the main reasons to have ULIDs, now we can insert entries in bulk and the IDs will be known.